src/main/java/com/marginallyclever/makelangelo/Makelangelo.java

```diff
@@ -1091,4 +1099,20 @@ public static void main(String[] args) {
 			makelangeloProgram.run();
 		});
 	}
-}
\ No newline at end of file
+	
+	public String getFullVersionString(){
+		return getLiteVersionString()+ " ("+DETAILED_VERSION+")";		
+	}
+	public String getLiteVersionString(){
+		return Translator.get("TitlePrefix") + " " +VERSION;		
+	}
+	
+	private static Makelangelo currentInstance = null;
+	
+	public static String staticgetFullVersionString(){
```

Comment:
the word static should not appear in the name of the method, we already know it is static.
If you have trouble with a method called in a static way and with instance, maybe you should put the code you want in a new object. Here, it could be a Version object wtih at least 2 static methods: getFullVersionString() and getLiteVersionString().
You should also put this new feature in a another branch as it is not related to the naming of turtle.

Reply:
Yes you are right I mixed several stories and one of them is currently very poorly implemented.
I wanted to have your opinions if this will be retained or not (named turtle, and JFrame Title with retail version only in dev mode, and Version in gcode ) because in each case there is work and I would not want to do it if it seems useless ... / unwanted

Reply:
The version part is very interesting.
I don't know about the named turtle, what will it be used for ?

Reply:
the idea of naming the turtles is to be able to propose a file name when you save (and to add the source file and any name + parameters values of the transformations applied as a comment) in the .gcode / . svg /.dxf output.)

A bit like with 3D printing slicers, you make it eat a "myFile".stl and it offers you when saving a "myFile"<possible other info, timestamp, ...>.gcode (and not like we currently have a file name to fill in to save ) and in the g-code you have in comment the ;MESH "name of the 3D object / name of the original file"